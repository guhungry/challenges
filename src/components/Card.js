import React, { PureComponent } from 'react';
import { Container, BackgroundImage, ContainerInfo, Overlay, ButtonClose } from './Card.Components'

import PropTypes from 'prop-types'

export default class Card extends PureComponent {
  static propTypes = {
    item: PropTypes.object,
    onPay: PropTypes.func
  }

  state = { selectedAmount: 0, showPayment: false };

  render () {
    const { item } = this.props
    const { showPayment } = this.state

    return (
      <Container>
        <BackgroundImage src={`/images/${item.image}`} />
        <ContainerInfo>
          <p>{item.name}</p>
          <button onClick={this.setShowPayment(true)}>Donate</button>
        </ContainerInfo>
        { showPayment && this.renderPaymentOverlay()}
      </Container>);
  }

  renderPaymentOverlay = () => {
    const { item, onPay } = this.props
    const { selectedAmount } = this.state
    const payments = [10, 20, 50, 100, 500].map((amount, j) => this.renderPayment(j, amount));

    return (
      <Overlay>
        <ButtonClose onClick={this.setShowPayment(false)}>X</ButtonClose>
        <div style={{textAlign: 'center'}}>
          <div>Select the amount to donate (USD)</div>
          <div>{ payments }</div>
          <button onClick={onPay.call(this, item.id, selectedAmount, item.currency)}>Pay</button>
        </div>
      </Overlay>
    )
  }

  renderPayment = (key, amount) => (
    <label key={key}>
      <input type="radio" name="payment" onClick={this.setAmount(amount)} /> {amount}
    </label>
  )

  setAmount = selectedAmount => () => this.setState({ selectedAmount })
  setShowPayment = showPayment => () => this.setState({ showPayment })
}