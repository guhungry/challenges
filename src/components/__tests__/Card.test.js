import React from 'react';
import Card from '../Card';
import renderer from 'react-test-renderer';

describe('Card', function() {
  test('should render currectly', function() {
    const component = renderer.create(
      <Card item={{ image: 'bee' }} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
