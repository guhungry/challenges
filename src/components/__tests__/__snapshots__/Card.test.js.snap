// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render currectly 1`] = `
<div
  className="sc-htpNat hnnook"
>
  <div
    className="sc-ifAKCX jRXylC"
  >
    <div
      className="sc-EHOje eUpySP"
      src="/images/bee"
    />
    <div
      className="sc-bZQynM bjIoQH"
    >
      <p />
      <div
        className="sc-bdVaJa KmHrp"
        disabled={false}
        onClick={[Function]}
      >
        Donate
      </div>
    </div>
  </div>
</div>
`;
